### PR TITLE
feat: erda-cli

### PR DESCRIPTION
**Pull request by jferic**
#### What this PR does / why we need it:
  implement erda-cli  clone/push/clone under project workspace

#### Specified Reviewers:

/assign @luobily @sixther-dc 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    implement erda-cli  clone/push/create under project workspace      |
| 🇨🇳 中文    |        围绕项目空间实现 erda-cli  的 clone/push/create 接口    |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: tools/cli/cmd/push.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		return errors.New("No application set to push.")
+	}
+
+	var org, project string
+	var orgId, projectId uint64
+
+	org = paths[1]
+	projectId, err = strconv.ParseUint(paths[4], 10, 64)
+
+	org, orgId, err = getOrgId(ctx, org, orgId)
```
Comment (luobily) [inline tools/cli/cmd/push.go:73]: The func design is not reasonable，same input and output

Comment (luobily) [reply tools/cli/cmd/push.go:73]: e.g. getOrgDetailByName、getOrgDetailByID ?

File: tools/cli/command/context.go [hunk 2]
```diff
@@ -45,16 +42,35 @@ func GetContext() *Context {
...
 	Name string `json:"name"`
 	Desc string `json:"desc"`
 }
 
+type ProjectInfo2 struct {
+	ID   uint64
+	Name string
+}
+
+type ApplicationInfo2 struct {
```
Comment (luobily) [inline tools/cli/command/context.go:64]: ApplicationInfo?

File: tools/cli/command/context.go [hunk 3]
```diff
@@ -45,16 +42,35 @@ func GetContext() *Context {
...
 	HttpClient         *httpclient.HTTPClient
 }
 type OrgInfo struct {
-	ID   int    `json:"id"`
+	ID   uint64 `json:"id"`
 	Name string `json:"name"`
 	Desc string `json:"desc"`
 }
 
+type ProjectInfo2 struct {
```
Comment (luobily) [inline tools/cli/command/context.go:59]: ProjectInfo?

File: tools/cli/command/context.go [hunk 4]
```diff
@@ -148,23 +134,23 @@ func (c *Context) AvailableOrgs() ([]apistructs.OrgDTO, error) {
...
+		envfilename = "erda_test.yml"
 	case "staging":
-		envfilename = "dice_staging.yml"
+		envfilename = "erda_staging.yml"
 	case "prod":
-		envfilename = "dice_production.yml"
+		envfilename = "erda_production.yml"
 	default:
-		envfilename = "dice.yml"
+		envfilename = "erda.yml"
```
Comment (luobily) [inline tools/cli/command/context.go:153]: Will changing to erda.yml make a difference?

File: tools/cli/common/project.go [hunk 5]
```diff
@@ -18,113 +18,270 @@ import (
...
+	if !resp.Success {
+		return apistructs.ProjectDTO{}, fmt.Errorf(utils.FormatErrMsg("get project detail",
+			fmt.Sprintf("failed to request, error code: %s, error message: %s",
+				resp.Error.Code, resp.Error.Msg), false))
+	}
+
+	return resp.Data, nil
+}
+
+func GetProjectIdByName(ctx *command.Context, orgId uint64, project string) (uint64, error) {
```
Comment (luobily) [inline tools/cli/common/project.go:67]:
```suggestion
func GetProjectIDByName(ctx *command.Context, orgId uint64, project string) (uint64, error) {
```

File: tools/cli/cmd/push.go [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+	Run: Push,
+}
+
+func Push(ctx *command.Context, urlstr string, applications []string, configfile string, pushall, force bool) error {
+	u, err := url.Parse(urlstr)
+	if err != nil {
+		return err
+	}
+	t, paths, err := utils.ClassifyURL(u.Path)
+
```
Comment (wangzhuzhen) [inline tools/cli/cmd/push.go:54]: 未做错误处理，后面 paths[] 引用可能越界

File: tools/cli/cmd/clone.go [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+	var pInfo *command.ProjectInfo
+	var successInfo string
+
+	_, _, err = command.GetProjectConfig()
+	if err != nil && err != utils.NotExist {
+		return err
+	} else if err == nil && t == utils.ProjectURL {
+		return errors.New("you are already in a erda project workspace.")
+	}
```
[the remark this answers is not in the document]
Comment (jferic) [reply tools/cli/cmd/clone.go:86]: 这里就是判断如果是 err == utils.NotExist 的情况 才继续往下

File: tools/cli/cmd/build.go [hunk 8]
```diff
@@ -15,92 +15,124 @@
...
+	return comps
+}
+
+// Create an pipeline and run it
+func PipelineRun(ctx *command.Context, filename, branch string, watch bool) error {
 	// 1. check if .git dir exists in current directory
 	// 2. parse current branch
 	// 3. create pipeline, run it
-	if _, err = os.Stat(".git"); err != nil {
+	if _, err := os.Stat(".git"); err != nil {
```
[the remark this answers is not in the document]
Comment (jferic) [reply tools/cli/cmd/build.go:104]: 嗯，先加强一下。虽然这个通过目录的方式也不是完全准确的

File: tools/cli/command/context.go (hunk 4 above)
Comment (jferic) [reply tools/cli/command/context.go:153]: 这部分代码没有用到，删了

File: tools/cli/common/project.go (hunk 5 above)
Comment (jferic) [reply tools/cli/common/project.go:67]: 这个部分代码没有用到，先去掉

File: tools/cli/cmd/build.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (wangzhuzhen) [reply tools/cli/cmd/build.go]: OK

File: tools/cli/cmd/clone.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (wangzhuzhen) [reply tools/cli/cmd/clone.go:89]: OK

**Pull request merged**